databases/Assignment2_answers.sql

```diff
@@ -0,0 +1,316 @@
+1. Which laptop models are shown in the Product relation?
+SELECT DISTINCT(Model)
+FROM Product
+WHERE Type = laptop;
+
+2. Find the set of all manufacturers ordered alphabetically.
+SELECT DISTINCT(Manufacturer)
+FROM Product
+ORDER BY DESC; # is there an alpha order?
+# all data types are by default order-able
+
+
+3. Which computer models appear in the database?
+SELECT Model
+FROM Product
+WHERE Type = pc OR Type = laptop;
+
+OR --> (below is a better idea lol)
+
+SELECT DISTINCT(Model)
+FROM Computer;
+
+
+4. Which computer models use 256 MB of RAM?
+SELECT Model
+FROM Computer
+WHERE Computer.RAM = 256;
+
+5. Which computer models cost less than $1100?
+SELECT Model
+FROM Computer
+WHERE Price < 1100;
+
+6. Which laptop models have a HD of at least 50 GB and cost less than $1400?
+SELECT Model
+FROM Computer
+INNER JOIN Product USING Model
+WHERE Product.Type = "laptop" Computer.HD >= 50 AND Price < 1400;
+
+7. Which laser printer models can print color documents?
+SELECT Model
+FROM Printer
+WHERE Printer.Type = laser AND Printer.Color = true;
+
+8. Which laptop models that have more than 256 MB of RAM
+are produced by manufacturer D?
+SELECT Computer.Model
+FROM Product
+INNER JOIN Computer USING Model
+WHERE Product.Type = laptop AND Computer.RAM > 256 AND Product.Manufacturer = D;
+
+9. Which computer models that have more than 256 MB of RAM
+are produced by manufacturer D?
+SELECT Computer.Model
+FROM Product
+INNER JOIN Computer USING Model
+WHERE Computer.RAM > 256 AND Product.Manufacturer = D;
+
+10. Which laptop models cost less than $1200?
+SELECT Computer.Model
+FROM Product
+INNER JOIN Computer USING Model
+WHERE Product.Type = laptop AND Computer.Price < 1200;
+
+11. Which manufacturers produce personal computers and printers?
+SELECT DISTINCT Manufacturer
+FROM Product
+WHERE Type = pc OR Type = printer;
```

Comment:
For a given row, Type can equal only one thing. This would return 0 rows.

You want to do a subquery for manufacturers who produce PCs and another subquery for manufacturers who produce printers, and then join them.